pkg/platform/yaml/yaml.go

```diff
@@ -188,3 +183,17 @@ func ReadSessionIndices(_ context.Context, path string, Logger *zap.Logger) ([]s
 	}
 	return indices, nil
 }
+
+func chmodRecursive(path string, mode os.FileMode) error {
```

Comment:
@charankamarapu instead of changing the permissions on the file, can't we set the right permissions while creating the file itself?

Reply:
that is already done - https://github.com/keploy/keploy/blob/aaa712e2e8751f9b42cad4937f05d12ed26972fe/pkg/platform/yaml/yaml.go#L149

Reply:
I have created the files and these are the permissions

<img width="1440" alt="Screenshot 2024-04-11 at 12 13 04 PM" src="https://github.com/keploy/keploy/assets/72094895/676dbe79-e025-4c78-9f44-a161ab3adbb1">

Reply:
can you please check the permissions for your folders..?

Reply:
that is weird.

Reply:
please delete the complete folder and record from scratch.

Reply:
I think the problem was in my laptop, I might have missed deleting the keploy folder properly. It worked as described after that.